orientdb/src/test/java/akka/stream/alpakka/orientdb/OrientDBTest.java

```diff
@@ -0,0 +1,328 @@
+    );
+
+    Collections.sort(result);
+    assertEquals(expect, result);
+  }
+
+  @Test
+  public void typedStream() throws Exception {
+    // Copy source/book to sink2/book through Typed stream
+    //#run-typed
+    CompletionStage<Done> f1 = OrientDBSource.typed(
+      source,
+      new OrientDBSourceSettings(oDatabase,
+        Runtime.getRuntime().availableProcessors(),
+        -1,
+        0,
+        10),
+      source1.class,
+      null
+    )
+      .map(m -> {
+        ODatabaseDocumentTx db = oDatabase.acquire();
+        db.setDatabaseOwner(new OObjectDatabaseTx(db));
+        ODatabaseRecordThreadLocal.instance().set(db);
+        sink2 sink = new sink2();
+        sink.setBook_title(m.oDocument().getBook_title());
+        return new Message<>(sink);
+      })
+      .runWith(OrientDBSink.typed(
+        sink2,
+        new OrientDBSinkSettings(oDatabase,
+          Runtime.getRuntime().availableProcessors(),
+          -1,
+          1,
+          5000,
+          10),
+        sink2.class
+      ), materializer);
+    //#run-typed
+
+    f1.toCompletableFuture().get(60, TimeUnit.SECONDS);
+
+    CompletionStage<List<String>> f2 = OrientDBSource.typed(
+      sink2,
+      new OrientDBSourceSettings(oDatabase,
+        Runtime.getRuntime().availableProcessors(),
+        -1,
+        0,
+        10),
+      sink2.class,
+      null
+    )
+      .map(m -> {
+        ODatabaseDocumentTx db = oDatabase.acquire();
+        db.setDatabaseOwner(new OObjectDatabaseTx(db));
+        ODatabaseRecordThreadLocal.instance().set(db);
+        return m.oDocument().getBook_title();
+      })
+      .runWith(Sink.seq(), materializer);
+
+    List<String> result = new ArrayList<>(f2.toCompletableFuture().get(60, TimeUnit.SECONDS));
+
+    List<String> expect = Arrays.asList(
+      "Akka Concurrency",
+      "Akka in Action",
+      "Effective Akka",
+      "Learning Scala",
+      "Programming in Scala",
+      "Scala Puzzlers",
+      "Scala for Spark in Production"
+    );
+
+    Collections.sort(result);
+    assertEquals(expect, result);
+  }
+
+  @Test
+  public void flow() throws Exception {
+    // Copy source to sink3 through ODocument stream
+    //#run-flow
+    CompletionStage<List<List<Message<ODocument>>>> f1 = OrientDBSource.create(
+      source,
+      new OrientDBSourceSettings(oDatabase,
+        Runtime.getRuntime().availableProcessors(),
+        -1,
+        0,
+        10),
+      null
+    )
+      .map(m -> new Message<>(m.oDocument()))
+      .via(OrientDBFlow.create(
+        sink3,
+        new OrientDBSinkSettings(oDatabase,
+          Runtime.getRuntime().availableProcessors(),
+          -1,
+          1,
+          5000,
+          10)
+      ))
+      .runWith(Sink.seq(), materializer);
+    //#run-flow
+
+    f1.toCompletableFuture().get();
+
+    // Assert docs in sink3
+    CompletionStage<List<String>> f2 = OrientDBSource.create(
+      sink3,
+      new OrientDBSourceSettings(oDatabase,

```

Comment:
Please create a companion object so that even Java users can get away with just setting the database.